src/smc-webapp/project_settings.cjsx

```diff
@@ -518,6 +528,37 @@ HideDeletePanel = rclass
                 This only impacts you, not your collaborators, and you can easily unhide it.
             </span>
 
+    render_delete_undelete_button : (is_deleted, is_expanded) ->
+        if is_deleted
+            text = "Undelete Project"
+            onClick = @toggle_delete_project
+            disabled = false
+        else
+            text = "Delete Project..."
+            onClick = @show_delete_conf
+            disabled = is_expanded
+
+        <Button bsStyle='danger' style={float: 'right'} onClick={onClick} disabled={disabled}>
+            <Icon name='trash' /> {text}
+        </Button>
+
+    render_expanded_delete_info : ->
+        <Well>
+            <Alert bsStyle="info" style={padding:'8px'} >
```

Comment:
Sorry to nitpick: can you change this to only get shown if the users has **actually applied** upgrades to this project?  If they haven't (usually the case?), it would be safe to just not show the alert, I think.